### ISSUE TITLE
NaN parameter after single backward with cuda_ad_rgb + prb on cbox.xml

### ISSUE THREAD

**Issue by 2545493686** [mitsuba-renderer/mitsuba3#1808]
**Summary**

I’m trying to run the gradient-based optimization example on the Cornell box scene. Even in a single-step minimal script (one render, one backward, one Adam step with lr=0), the optimized parameter becomes NaN on GPU.

**System information**

> OS: Windows 11 x64
GPU: RTX 4070 TI Super
NVIDIA driver: 560.94
Python: 3.10.19 (conda)
Mitsuba: 3.7.1
DrJit: 1.2.0
Install method: pip install mitsuba

**Minimal reproducer**

```python
import drjit as dr
import mitsuba as mi

mi.set_variant('cuda_ad_rgb')

scene = mi.load_file(r'G:\AI\studio\POSTM\data\mitsuba_scene\cbox.xml', res=128, integrator='prb')
image_ref = mi.render(scene, spp=512)

params = mi.traverse(scene)

key = 'red.reflectance.value'

# Save the original value
param_ref = mi.Color3f(params[key])

# Set another color value and update the scene
params[key] = mi.Color3f(0.01, 0.2, 0.9)
params.update()

opt = mi.ad.Adam(lr=0)
opt[key] = params[key]
params.update(opt)

def mse(image):
    return dr.mean(dr.square(image - image_ref))

# Perform a (noisy) differentiable rendering of the scene
image = mi.render(scene, params, spp=64, seed=43)

# Evaluate the objective function from the current rendered image
loss = mse(image)

# Backpropagate through the rendering process
dr.backward(loss)

# Optimizer: take a gradient descent step
opt.step()
# Update the scene state to the new optimized values
params.update(opt)

# Post-process the optimized parameters to ensure legal color values.
print(f"loss = {loss}, current color = {params[key]}")

```

**Observed output**
```powershell
python .\mi_origin.py
(postm) PS G:\AI\studio\POSTM\tools\mi_origin> python .\mi_origin.py
jitc_llvm_init(): LLVM API initialization failed ..
loss = 0.00600782, current color = [[nan, nan, nan]]
```

**Additional observations**

- The optimized parameter is a `Color3f`. After a single backward + Adam step with `lr=0`, the parameter becomes `[NaN, 0.2, NaN]`: the R and B channels are NaN, while the G channel remains exactly 0.2.
- This behavior is reproducible when `dr.set_flag(dr.JitFlag.Debug, True)` is enabled. The loss itself is finite (`~0.006`), so the NaN seems to come from the gradient or optimizer update, not from the MSE computation.

```powershell
python .\mi_origin.py
(postm) PS G:\AI\studio\POSTM\tools\mi_origin> python .\mi_origin.py
jitc_llvm_init(): LLVM API initialization failed ..
loss = 0.00600782, current color = [[nan, 0.2, nan]]
```

**Comment by lnuic** [comment]
Dear @2545493686,
this bug might be related to issue https://github.com/mitsuba-renderer/mitsuba3/pull/1818.
Could you please check whether that solves your issue?

**Comment by njroussel** [comment]
Closing.

I can't reproduce this and we haven't heard back from @2545493686
